core/src/main/java/com/google/errorprone/bugpatterns/StringSplit.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2017 Two Sigma Open Source. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.errorprone.bugpatterns;
+
+import static com.google.errorprone.BugPattern.Category.JDK;
+import static com.google.errorprone.BugPattern.SeverityLevel.WARNING;
+import static com.google.errorprone.matchers.Description.NO_MATCH;
+import static com.google.errorprone.matchers.method.MethodMatchers.instanceMethod;
+
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker.MethodInvocationTreeMatcher;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.MethodInvocationTree;
+import java.util.List;
+
+/** @author dturner@twosigma.com (David Turner) */
+@BugPattern(
+  name = "StringSplit",
+  category = JDK,
+  summary = "String.split should never take only a single argument; it has surprising behavior",
+  severity = WARNING
+)
+public class StringSplit extends BugChecker implements MethodInvocationTreeMatcher {
```

Comment:
The check needs to be added to this list to get wired up: https://github.com/google/error-prone/blob/2623f228cece3a24fd7ccb77be478c4e81f2883d/core/src/main/java/com/google/errorprone/scanner/BuiltInCheckerSuppliers.java#L525